internal/backup/volumesnapshotcontent_action.go

```diff
@@ -73,9 +87,72 @@ func (p *VolumeSnapshotContentBackupItemAction) Execute(item runtime.Unstructure
 
 	snapContMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&snapCont)
 	if err != nil {
-		return nil, nil, errors.WithStack(err)
+		return nil, nil, "", nil, errors.WithStack(err)
+	}
+
+	// The operationID is of the form <volumesnapshotcontent-name>/<started-time>
+	operationID := snapCont.Name + "/" + time.Now().Format(time.RFC3339)
+	itemToUpdate := []velero.ResourceIdentifier{
+		{
+			GroupResource: kuberesource.VolumeSnapshotContents,
+			Name:          snapCont.Name,
+		},
 	}
 
 	p.Log.Infof("Returning from VolumeSnapshotContentBackupItemAction with %d additionalItems to backup", len(additionalItems))
-	return &unstructured.Unstructured{Object: snapContMap}, additionalItems, nil
+	return &unstructured.Unstructured{Object: snapContMap}, additionalItems, operationID, itemToUpdate, nil
+}
+
+func (p *VolumeSnapshotContentBackupItemAction) Name() string {
+	return "VolumeSnapshotContentBackupItemAction"
+}
+
+func (p *VolumeSnapshotContentBackupItemAction) Progress(operationID string, backup *velerov1api.Backup) (velero.OperationProgress, error) {
+	progress := velero.OperationProgress{}
+	if operationID == "" {
+		return progress, biav2.InvalidOperationIDError(operationID)
+	}
+
+	// The operationId is of the form <volumesnapshotcontent-name>/<started-time>
+	operationsIDParts := strings.Split(operationID, "/")
+	if len(operationsIDParts) != 2 {
+		p.Log.WithField("operationID", operationID).Error("Invalid operationID")
+		return progress, biav2.InvalidOperationIDError(operationID)
+	}
+	var err error
+	if progress.Started, err = time.Parse(time.RFC3339, operationsIDParts[1]); err != nil {
+		p.Log.Errorf("error parsing operationID's StartedTime part into time %s: %s", operationID, err.Error())
+		return progress, errors.WithStack(fmt.Errorf("fail to parse StartedTime: %s", err.Error()))
+	}
+
+	_, snapshotClient, err := util.GetClients()
+	if err != nil {
+		return progress, errors.WithStack(err)
+	}
+
+	vsc, err := snapshotClient.SnapshotV1().VolumeSnapshotContents().Get(context.Background(), operationsIDParts[0], metav1.GetOptions{})
+	if err != nil {
+		p.Log.Errorf("error getting volumesnapshotcontent %s: %s", operationsIDParts[0], err.Error())
+		return progress, errors.WithStack(err)
+	}
+
+	if vsc.Status == nil {
+		return progress, nil
```

Comment:
maybe add a debug log here that VSC is yet to have a status or something

Reply:
Done